sympy/core/evalf.py

```diff
@@ -1473,7 +1473,11 @@ def evalf(self, n=15, subs=None, maxn=100, chop=False, strict=False, quad=None,
             result = evalf(self, prec + 4, options)
         except NotImplementedError:
             # Fall back to the ordinary evalf
-            v = self._eval_evalf(prec)
+            from sympy.core.basic import Basic
+            if isinstance(self, Basic) and subs is not None:  # issue 20291
```

Comment:
I think it should not be isinstance(..., Basic), because Classes in Physics module are not Basic and .evalf() is being added to them.

It should rather be 

if subs is not None:
    try:
        v = self.subs(subs)._eval_evalf(prec)
    except Error:
        .......

Reply:
Will do. I know @oscarbenjamin does not really like try catching so I think I'll do `hasattr('_evalf')` instead.

Reply:
Okay!